### PR TITLE
SITES-916: Put sites in maintenance mode for final cutover

### PR DESCRIPTION
**Pull request by jbickar**
# READY FOR REVIEW

# Summary
- Put both the Sites website and the ACSF website into maintenance mode while content migration is underway
- Take them out when finished

# Needed By (Date)
- 9.28.18

# Criticality
- How critical is this PR on a 1-10 scale? 5/10

# Steps to Test

1. Check out this branch
2. Run `full-migration-playbook.yml` on a site
3. Once it gets to the beginning of the `download-site` role, check that both the Sites (or People) site *and* the ACSF site are in maintenance mode
4. Once it gets to the end of the `change-paths` role, verify that both the Sites (or People) site *and* the ACSF site are out of maintenance mode
5. Run `migration-sync-only-playbook.yml` on the same site
6. Repeat steps 3 and 4

# Affected Projects or Products
- ACSF

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-916

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by jbickar** [comment]
I had some issues with this so you can hold off until tomorrow to test this.

**Comment by sherakama** [comment]
Two questions come to mind

1. Why would we need to put the site into maintenance mode on ACSF? What benefit do we get from this?
2. Why do we take the sites-1.0 site out of maintenance mode after it has been moved? Wouldn't we want to keep the users locked out until the full migration has completed and the vhosts switched over?

**Comment by jbickar** [comment]
This functionality is the flip side of https://stanfordits.atlassian.net/browse/SITES-930 and https://stanfordits.atlassian.net/browse/SITES-931.

I would *like to* put all sites in maintenance mode on both platforms during the final content sync. I'm still not sure that will be feasible, so this PR is to give another option. I suppose it should be tied to a variable, then. 🤔

File: roles/change-paths/tasks/main.yml [hunk 1]
```diff
@@ -193,3 +193,10 @@
 - name: Save modified JS Injector rules to disk
   shell: "{{ drush_alias }} scr {{ scripts_dir }}/js-injector.php"
   notify: Clear site cache
+
+# Bring 'er out of hyperspace, Chewy
+- name: Turn off maintenance mode for Sites 1.x site and ACSF site
+  shell: "{{ item }} -y --exact vset maintenance_mode 0"
+  with_items:
+    - " {{ drush_alias }}"
+    - " drush @{{ server_alias }}.{{ site_prefix }}_{{ inventory_hostname }} "
```
Comment (sherakama) [inline roles/change-paths/tasks/main.yml]: I can adjust my aliases much easier than trying to adjust this code. I'll change it, but it is a good lesson learned for future scripts is that we cannot assume each other's aliases.

**Comment by sherakama** [comment]
Ok, this code looks fine to me. I am still not sure what the gains are by doing this other than locking users out of the current production site while migration happens.

File: roles/change-paths/tasks/main.yml (hunk 1 above)
Comment (jbickar) [reply roles/change-paths/tasks/main.yml]: Yeah there are a few things at play here, lessons learned over a year of iterating on this software package.

1. We should have uniquely namespaced the drush aliases (for both on-prem and ACSF) that these Ansible playbooks use.
2. I have inadvertently been pushing a transition away from OS X and one's everyday development environments, to `my-stanford-vbox` and `sites-uat`, which are Ubuntu/Debian, and more purposefully tuned for this work.

Sorry for the friction.

Comment (jbickar) [reply roles/change-paths/tasks/main.yml]: Oh. Yeah.

https://github.com/SU-SWS/ansible-playbooks/blob/master/roles/download-site/tasks/main.yml#L39

I can add that in here.

**Comment by jbickar** [comment]
Given the fact that I can't yet get a reliable benchmark on how long the content migration for 800 sites is actually going to take, that's enough of a benefit for me. 

However, there are two changes that I need to make:
1. Make this dependent on a variable (if Karen is able to turn on the maintenance mode for the on-prem service as planned, then we won't need this)
2. Support non-default drush alias syntax.

**Comment by jbickar** [comment]
1. We can't assume that the site being migrated **wants** to come out of maintenance mode.
2. Even if this is conditional on a variable, I don't think we want the overhead of 4 skipped tasks per host.

**Comment by jbickar** [comment]
I'm going to come back to this for the sites1/sites2 non-vhost cutover. We do not have the option of doing the server-level maintenance mode for that.

**Pull request closed without merging**